### DISCUSSION TITLE
Bug: Gluetun shuts down as soon as I start it

### DISCUSSION THREAD

**Discussion by Smalfinn** [qdm12/gluetun#670] [Q&A]
### Is this urgent?

No

### Host OS

My Cloud OS 5 based on Debian 10

### CPU arch

x86_64

### VPN service provider

Private Internet Access

### What are you using to run the container

docker run

### What is the version of Gluetun

Running version latest built on 2021-10-06T18:59:13Z (commit caeca18)

### What's the problem 🤔

Gluetun shuts down as soon as I start it and I don't know what I can do about it.

### Share your logs

```log
========================================
========================================
=============== gluetun ================
========================================
=========== Made with ❤️ by ============
======= https://github.com/qdm12 =======
========================================
========================================

Running version latest built on 2021-10-06T18:59:13Z (commit caeca18)

🔧 Need help? https://github.com/qdm12/gluetun/discussions/new
🐛 Bug? https://github.com/qdm12/gluetun/issues/new
✨ New feature? https://github.com/qdm12/gluetun/issues/new
☕ Discussion? https://github.com/qdm12/gluetun/discussions/new
💻 Email? quentin.mcgaw@gmail.com
💰 Help me? https://www.paypal.me/qmcgaw https://github.com/sponsors/qdm12
2021/10/12 13:50:45 INFO storage: creating /gluetun/servers.json with 11225 hard coded servers
2021/10/12 13:50:45 INFO Alpine version: 3.14.2
2021/10/12 13:50:45 INFO OpenVPN 2.4 version: 2.4.11
2021/10/12 13:50:45 INFO OpenVPN 2.5 version: 2.5.2
2021/10/12 13:50:45 INFO Unbound version: 1.13.2
2021/10/12 13:50:45 INFO IPtables version: v1.8.7
2021/10/12 13:50:45 INFO Settings summary below:
|--VPN:
   |--Type: openvpn
   |--OpenVPN:
      |--Version: 2.5
      |--Verbosity level: 1
      |--Network interface: tun0
      |--Encryption preset: strong
   |--Private Internet Access settings:
      |--Regions: uk london
      |--OpenVPN selection:
         |--Protocol: udp
         |--PIA encryption preset: strong
|--DNS:
   |--Plaintext address: 1.1.1.1
   |--DNS over TLS:
      |--Unbound:
          |--DNS over TLS providers:
              |--Cloudflare
          |--Listening port: 53
          |--Access control:
              |--Allowed:
                  |--0.0.0.0/0
                  |--::/0
          |--Caching: enabled
          |--IPv4 resolution: enabled
          |--IPv6 resolution: disabled
          |--Verbosity level: 1/5
          |--Verbosity details level: 0/4
          |--Validation log level: 0/2
          |--Username:
      |--Blacklist:
         |--Blocked categories: malicious
         |--Additional IP networks blocked: 13
      |--Update: every 24h0m0s
|--Firewall:
|--Log:
   |--Level: INFO
|--System:
   |--Process user ID: 1000
   |--Process group ID: 1000
   |--Timezone: europe/london
|--Health:
   |--Server address: 127.0.0.1:9999
   |--Address to ping: github.com
   |--VPN:
      |--Initial duration: 6s
      |--Addition duration: 5s
|--HTTP control server:
   |--Listening port: 8000
   |--Logging: enabled
|--Public IP getter:
   |--Fetch period: 12h0m0s
   |--IP file: /tmp/gluetun/ip
|--Github version information: enabled
2021/10/12 13:50:45 INFO routing: default route found: interface eth0, gateway 172.17.0.1
2021/10/12 13:50:45 INFO routing: local ethernet link found: eth0
2021/10/12 13:50:45 INFO routing: local ipnet found: 172.17.0.0/16
2021/10/12 13:50:45 INFO routing: default route found: interface eth0, gateway 172.17.0.1
2021/10/12 13:50:45 INFO routing: default route found: interface eth0, gateway 172.17.0.1
2021/10/12 13:50:45 INFO routing: deleting route for 0.0.0.0/0
2021/10/12 13:50:45 ERROR routing: cannot reverse routing changes: cannot remove routes for inbound traffic from default IP: cannot delete rule: cannot get default IP address: default link not found: in 1 route(s)
2021/10/12 13:50:45 ERROR cannot setup routing: cannot add routes for inbound traffic from default IP: cannot add rule: cannot add rule: cannot list rules: operation not supported
2021/10/12 13:50:45 INFO Shutdown successful
```

### Share your configuration

```yml
docker run -it --rm --cap-add=NET_ADMIN -e VPNSP="private internet access" \
-e OPENVPN_USER=*removed* -e OPENVPN_PASSWORD=*removed* \
-v gluetun:/gluetun \
-e REGION=Netherlands qmcgaw/gluetun
```

**Comment by qdm12** [comment]
Does this work?

```sh
docker run -it --rm alpine:3.14 ip rule list
```

Maybe it's something missing in your kernel to handle ip rules 🤔

**Comment by Smalfinn** [comment]
> Does this work?
> 
> ```shell
> docker run -it --rm alpine:3.14 ip rule list
> ```
> 
> Maybe it's something missing in your kernel to handle ip rules 🤔

This was the outcome - 
```shell
root@MyCloudPR4100 ~ # docker run -it --rm alpine:3.14 ip rule list
Unable to find image 'alpine:3.14' locally
3.14: Pulling from library/alpine
a0d0a0d46f8b: Already exists
Digest: sha256:e1c082e3d3c45cccac829840a25941e679c25d438cc8412c2fa221cf1a824e6a
Status: Downloaded newer image for alpine:3.14
ip: RTNETLINK answers: Not supported
ip: dump terminated
```

**Comment by qdm12** [comment]
Your kernel is missing the iproute module. Try with `apt-get update -y && apt-get install -y iproute2` and re-run that Docker command above to see if it works?

**Comment by Smalfinn** [comment]
> Your kernel is missing the iproute module. Try with `apt-get update -y && apt-get install -y iproute2` and re-run that Docker command above to see if it works?

```shell
-sh: apt-get: not found
```

Uh-oh

**Comment by qdm12** [comment]
Ah it's one of those manufacturer stripped down OSes 😢
And it does run Docker? That's crazy 😄

Maybe the kernel module is there but it's not loaded. Try to load it with:

```sh
sudo modprobe iproute
echo $?
```

It should show `0` on success and something else otherwise.

EDIT: I'll convert it to a discussion since this is more of a host kernel issue than a gluetun bug.

**Reply by Smalfinn** [reply]
> Ah it's one of those manufacturer stripped down OSes 😢 And it does run Docker? That's crazy 😄
> 
> Maybe the kernel module is there but it's not loaded. Try to load it with:
> 
> ```shell
> sudo modprobe iproute
> echo $?
> ```
> 
> It should show `0` on success and something else otherwise.
> 
> EDIT: I'll convert it to a discussion since this is more of a host kernel issue than a gluetun bug.

```shell
root@MyCloudPR4100 ~ # sudo modprobe iproute
modprobe: can't change directory to '/lib/modules': No such file or directory
root@MyCloudPR4100 ~ # echo $?
1
root@MyCloudPR4100 ~ #
```
Looks like I don't have it

**Reply by qdm12** [reply]
Oh boy, I would suggest to reach out to the manufacturer and ask how you can install the `iproute` kernel module in your system. Maybe can you upgrade your system?

**Reply by Smalfinn** [reply]
No upgrades pending.

I'll contact support and see if they can assist, might be worth me installing unraid or something instead of the stripped down manufacturers' OS

**Reply by Smalfinn** [reply]
Support were useless - 

> I regret to inform you that we have not tested the My cloud drive with a VPN server.
> Due to the fact that My Cloud products are tested in simple networks, we do not offer support on any VPN related issues, since it requires specialized network configuration.
> We do apologize for any inconvenience this may have caused.
> If you have not already done so, please consult OpenVPN, any IT administrator (if applicable) and/or your ISP.
> If you experience any issues further or require any type of assistance, please feel free to reply to this email.
> Thank you for choosing Western Digital Support. Have a nice day.

I guess MyCloud PR4100 weren't meant to have VPN support

**Reply by qdm12** [reply]
Don't mention the VPN, just ask them you need to be able to interact with `ip rule` and right now it gives you the error message `Operation is not supported`. Ip route doesn't work either (try it on the host `ip route`)?

**Reply by jasperzs** [reply]
I also face this issue on MyCloudEX2Ultra. Did you eventually fix this issue?

**Reply by JulienPreuilh** [reply]
@qdm12 , reviving this a bit as I am facing this issue myself. ip route works, here's the result : 
default via 192.168.0.1 dev bond0
172.17.0.0/16 dev docker0  proto kernel  scope link  src 172.17.0.1
172.21.0.0/16 dev br-f655b08bfec4  proto kernel  scope link  src 172.21.0.1
172.22.0.0/16 dev br-32e36c14780d  proto kernel  scope link  src 172.22.0.1
192.168.0.0/24 dev bond0  proto kernel  scope link  src 192.168.0.248


Western digital NASes run busybox, which is striped of a lot of functionalities it seems. I am half way convinced I need to change the OS (which is not a straighforward process at all)
